jupyterhub/templates/_helpers-names.tpl

```diff
@@ -196,11 +196,29 @@
     {{- include "jupyterhub.fullname.dash" . }}hook-image-puller
 {{- end }}
 
+{{- /* hook-image-puller ServiceAccount */}}
+{{- define "jupyterhub.hook-image-puller-serviceaccount.fullname" -}}
+    {{- if .Values.prePuller.hook.serviceAccountImagePuller.create }}
+        {{- .Values.prePuller.hook.serviceAccountImagePuller.name | default (include "jupyterhub.hook-image-puller.fullname" .) }}
+    {{- else }}
+        {{- .Values.prePuller.hook.serviceAccountImagePuller.name | default "default" }}
+    {{- end }}
+{{- end }}
+
 {{- /* continuous-image-puller DaemonSet */}}
 {{- define "jupyterhub.continuous-image-puller.fullname" -}}
     {{- include "jupyterhub.fullname.dash" . }}continuous-image-puller
 {{- end }}
 
+{{- /* continuous-image-puller ServiceAccount */}}
+{{- define "jupyterhub.continuous-image-puller-serviceaccount.fullname" -}}
+    {{- if .Values.prePuller.continuous.serviceAccount.create }}
+        {{- .Values.prePuller.continuous.serviceAccount.name | default (include "jupyterhub.continuous-image-puller.fullname" .) }}
+    {{- else }}
+        {{- .Values.prePuller.continuous.serviceAccount.name | default "default" }}
+    {{- end }}
+{{- end }}
```

Comment:
This mimiced some old practice, but I figure it doesn't make sense at least for these resources, so I didn't follow the practice here, and made this into::

```
{{- /* continuous-image-puller ServiceAccount */}}
{{- define "jupyterhub.continuous-image-puller-serviceaccount.fullname" -}}
    {{- .Values.prePuller.continuous.serviceAccount.name | default (include "jupyterhub.continuous-image-puller.fullname" .) }}
{{- end }}
```